### DISCUSSION TITLE
GNU Radio - PlutoSDR

### DISCUSSION THREAD

**Discussion by EvpSeven** [duggabe/gr-control#8] [Q&A] [answered]
Hello duggabe,

It is possible to adapt your program to use a PLUTO-SDR?
I tried modifying pkt_rcv.grc and pkt_xmt.grc to contain the PLUTO-SDR blocks. It does not appear to receive anything, as the correlate out remains at zero.

Sorry for bothering,
Best regards EvpSeven.

**Comment by duggabe** [comment, marked as answer]
The recommended approach is to use three processes (terminal windows): xmt_rcv_switch_Pluto, pkt_rcv, and pkt_xmt as described in the README.md. Are you using GNU Radio 3.10?

**Reply by EvpSeven** [reply]
Yes iam using 3.10 version

**Reply by EvpSeven** [reply]
The idea is transmitted image in pluto and received in another Pluto, in different PC.

**Reply by duggabe** [reply]
My recommendation is exactly what I have been doing: all three tasks running in both computers.
If you wanted to do so, you could create a simplified xmt_rcv_switch_Pluto for the transmit side that just had the ZMQ SUB Source and the PlutoSDR Sink and leave out the switching logic.

**Comment by EvpSeven** [comment]
I followed the tutorial on the wiki (https://wiki.gnuradio.org/index.php?title=File_transfer_using_Packet_and_BPSK#Sending_Non-text_Files), and it works correctly internally on gnu radio with the ZMQ blocks. I tried to implement it with the pluto sink and source blocks and it doesn't work.

**Reply by duggabe** [reply]
You have to click 'Transmit' before you can execute the pkt_xmt!

**Comment by Dio76453** [comment]
Hi!
I'm working with EvpSeven on an university project and our task is to send an image from a raspberry pi inside a payload of a UAV/HAB and a laptop as a ground station, using two SDR for the data transfer and both of us are touching gnu-radio for the first time. The idea is to transmit with GFSK modulation, but I stumbled on that wiki and it sent me to this repository, which is the most recent (that I could find) and it uses packet communication and signal modulation. So to test it I only followed the wiki and just changed the ports of the ZMQ block on both xmt and rcv and I could receive the data, remove the preamble and decode it with strip_preamble.py. The next step is to transmit and receive with the PLUTOs and then try to change the modulation type.

**Reply by duggabe** [reply]
In addition to consolidating the RF portion on one place, the UHD version of xmt_rcv_switch was created because only one process (flowgraph) can have ownership of the source and sink blocks at the same time. That may not be the case for Plutos.

For your use case, it's probably simpler just to replace the ZMQ source with a Pluto source in pkt_rcv; and replace the ZMQ sink with a Pluto sink in the pkt_xmt.

Make sure you set an appropriate frequency, such as in an ISM band. During testing, I have found that the receive tuning is very critical and doesn't exactly match the transmit frequency. If you have a moving transmitter, that further complicates the tuning!

**Reply by Dio76453** [reply]
That's what I did in this next step that I described. Tomorrow I'll try to tune the frequency and see if I have better results.
Thanks!

**Reply by duggabe** [reply]
I have updated pkt_rcv to add a FLL Band-Edge block and an AGC. That greatly improves the tuning problem.
Just go to $HOME/gr-control and do a "git pull" to update your copy.

**Reply by Dio76453** [reply]
I'll check it now! Thanks a lot!
Also I joined the GNU Radio General chat just now

**Comment by duggabe** [comment]
@EvpSeven and @Dio76453: Can you join the GNU Radio General Chat channel? See https://wiki.gnuradio.org/index.php?title=Chat
It would make this discussion a lot easier than here. Thanks.
